src/components/CreateMenu/BaseCreateMenu.js

```diff
@@ -0,0 +1,86 @@
+import React, {PureComponent} from 'react';
+import {View} from 'react-native';
+import PropTypes from 'prop-types';
+import Popover from '../Popover';
+import styles from '../../styles/styles';
+import withWindowDimensions, {windowDimensionsPropTypes} from '../withWindowDimensions';
+import MenuItem from '../MenuItem';
+
+const propTypes = {
+    // Callback to fire on request to modal close
+    onClose: PropTypes.func.isRequired,
+
+    // State that determines whether to display the create menu or not
+    isVisible: PropTypes.bool.isRequired,
+
+    // Callback to fire when a CreateMenu item is selected
+    onItemSelected: PropTypes.func.isRequired,
+
+    // Menu items to be rendered on the list
+    menuItems: PropTypes.arrayOf(
+        PropTypes.shape({
+            icon: PropTypes.func.isRequired,
+            text: PropTypes.string.isRequired,
+            onSelected: PropTypes.func.isRequired,
+        }),
+    ).isRequired,
+
+    /* Configures when menu item actions are triggered: as soon as pressed or after the modal is closed.
+    * On mobile native we need to wait for the modals to close, while on web as soon as clicked */
+    actionExecutionStrategy: PropTypes.oneOf(['ON_PRESS', 'AFTER_MODAL_CLOSE']),
```

Comment:
I like this idea (much better than two separate props). Can you please make these constants that live in `CONST.js`?

Reply:
Actually taking a new look and I'll need to change this one last time for the best. I'll post a note on the code below